plugins/scenario/hu.bme.mit.gamma.scenario.statechart.generator/src/hu/bme/mit/gamma/scenario/statechart/generator/AbstractContractStatechartGeneration.xtend

```diff
@@ -0,0 +1,505 @@
+import hu.bme.mit.gamma.statechart.interface_.InterfaceModelFactory
+import hu.bme.mit.gamma.statechart.interface_.Port
+import hu.bme.mit.gamma.statechart.interface_.TimeUnit
+import hu.bme.mit.gamma.statechart.interface_.Trigger
+import hu.bme.mit.gamma.statechart.statechart.BinaryTrigger
+import hu.bme.mit.gamma.statechart.statechart.BinaryType
+import hu.bme.mit.gamma.statechart.statechart.ChoiceState
+import hu.bme.mit.gamma.statechart.statechart.Region
+import hu.bme.mit.gamma.statechart.statechart.State
+import hu.bme.mit.gamma.statechart.statechart.StateNode
+import hu.bme.mit.gamma.statechart.statechart.StatechartDefinition
+import hu.bme.mit.gamma.statechart.statechart.StatechartModelFactory
+import hu.bme.mit.gamma.statechart.statechart.Transition
+import hu.bme.mit.gamma.statechart.statechart.UnaryTrigger
+import hu.bme.mit.gamma.statechart.statechart.UnaryType
+import hu.bme.mit.gamma.statechart.util.StatechartUtil
+import hu.bme.mit.gamma.util.GammaEcoreUtil
+import hu.bme.mit.gamma.util.JavaUtil
+import java.util.HashMap
+import java.util.List
+import java.util.Map
+
+import static extension hu.bme.mit.gamma.statechart.derivedfeatures.StatechartModelDerivedFeatures.*
+
+abstract class AbstractContractStatechartGeneration {
+
+	protected val extension StatechartModelFactory statechartfactory = StatechartModelFactory.eINSTANCE
+	protected val extension ExpressionModelFactory expressionfactory = ExpressionModelFactory.eINSTANCE
+	protected val extension InterfaceModelFactory interfacefactory = InterfaceModelFactory.eINSTANCE
+	protected val extension ActionModelFactory actionfactory = ActionModelFactory.eINSTANCE
+	protected val extension ContractModelFactory contractfactory = ContractModelFactory.eINSTANCE
+	protected val extension GammaEcoreUtil ecoreUtil = GammaEcoreUtil.INSTANCE
+	protected val extension ExpressionEvaluator exprEval = ExpressionEvaluator.INSTANCE
+	protected val extension ExpressionUtil exprUtil = ExpressionUtil.INSTANCE
+	protected val extension ScenarioStatechartUtil scenarioStatechartUtil = ScenarioStatechartUtil.INSTANCE
+	protected val extension ScenarioModelDerivedFeatures scenarioModelDerivedFeatures = ScenarioModelDerivedFeatures.
+		INSTANCE
+	protected val StatechartUtil statechartUtil = StatechartUtil.INSTANCE
+
+	protected val JavaUtil javaUtil = JavaUtil.INSTANCE
+
+	protected var Component component = null
+	protected var ScenarioDefinition scenario = null
+	protected var StatechartDefinition statechart = null
+	protected val variableMap = <String, VariableDeclaration>newHashMap
+	protected var exsistingChoices = 0
+	protected var exsistingMerges = 0
+	protected var stateCount = 0
+	protected var timeoutCount = 0
+	protected var Region firstRegion = null
+	protected var StateNode previousState = null
+	protected var State hotViolation = null
+	protected var State coldViolation = null
+	protected val replacedStateWithValue = new HashMap<StateNode, StateNode>()
```

Comment:
Maybe `Map<StateNode, StateNode> replacedStateWithValue = newHashMap` is more readable. Make it consistent with `val variableMap = <String, VariableDeclaration>newHashMap`